### DISCUSSION TITLE
Express support

### DISCUSSION THREAD

**Discussion by fbritoferreira** [asteasolutions/zod-to-openapi#205] [Announcements]
Would it be a good idea to add a few helpers to make it easier to use this in express?
It would be nice if we could take a schema and make sure the path of the express handler is validated against it also params etc.
Happy to work on this if you think it would add value.

**Comment by AGalabov** [comment]
@fbritoferreira we've definitely thought of that and we've kind of implemented such a mechanism in our application but we've never really created it as a reusable library/framework. Not sure when we would get the time for such an effort. There is already https://hono.dev/ that has integrated with our library so something along those lines can definitely be implemented as a project for Express.

I'd turn this into a discussion for now

**Reply by fbritoferreira** [reply]
Can you share some details on how hono integrates? I may migrate to it if easier to use.

**Comment by evertdespiegeleer** [comment]
There are a few projects which _kinda_ do something similar to what you're describing. As I was looking for something similar and nothing really gave me the flexibility I wanted, I actually started work on a lib based on express and zod-to-openapi a few days ago to accomplish exactly this. Not very well documented yet, but hope it helps :) https://github.com/evertdespiegeleer/zhttp

**Reply by fbritoferreira** [reply]
This seems cool going to try and use it and see how it goes
